FreydCategoriesForCAP/gap/QuiverRows.gi

```diff
@@ -1335,7 +1291,15 @@ InstallGlobalFunction( INSTALL_FUNCTIONS_FOR_QUIVER_ROWS,
             
         od;
         
-        return morphism_constructor( distinguished_object, row, object_constructor( Size( row ) ) );
+        if IsEmpty( row ) then
+            
+            return ZeroMorphism( hom_structure_range_category, source, range );
```

Comment:
Is this line the reason why you needed to introduce the new WithGiven version of this function?

Reply:
Technically, introducing the new WithGiven version was not a "must", but yes, that's where it comes in handy. I already thought about this yesterday when writing the following lines of code: https://github.com/homalg-project/Algebroids/blob/6eb2a9884facf904e3b76d5f07123f320c7174f5/gap/Algebroids.gi#L731-L732 And I think it's natural to have the WithGiven version, or is there anything that distinguishes this case from e.g. `HomomorphismStructureOnMorphismsWithGivenObjects`?